pkg/controller/agent/nodenetwork/controller.go

```diff
@@ -0,0 +1,354 @@
+package nodenetwork
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/vishvananda/netlink"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog"
+
+	networkv1alpha1 "github.com/rancher/harvester-network-controller/pkg/apis/network.harvester.cattle.io/v1alpha1"
+	"github.com/rancher/harvester-network-controller/pkg/config"
+	"github.com/rancher/harvester-network-controller/pkg/controller/common"
+	"github.com/rancher/harvester-network-controller/pkg/generated/controllers/network.harvester.cattle.io/v1alpha1"
+	"github.com/rancher/harvester-network-controller/pkg/network"
+	"github.com/rancher/harvester-network-controller/pkg/network/iface"
+	"github.com/rancher/harvester-network-controller/pkg/network/vlan"
+
+	harvnetwork "github.com/rancher/harvester/pkg/api/network"
+
+	cniv1 "github.com/rancher/harvester/pkg/generated/controllers/k8s.cni.cncf.io/v1"
+)
+
+// NodeNetwork controller watches NodeNetwork to configure network for cluster node
+const (
+	controllerName = "nodenetwork-controller"
+	resetPeriod    = time.Minute * 30
+
+	bridgeCNIName = "bridge"
+)
+
+type Handler struct {
+	nodeNetworkCtr   v1alpha1.NodeNetworkController
+	nodeNetworkCache v1alpha1.NodeNetworkCache
+	nadCache         cniv1.NetworkAttachmentDefinitionCache
+	recorder         record.EventRecorder
+}
+
+func Register(ctx context.Context, management *config.Management) error {
+	nns := management.HarvesterNetworkFactory.Network().V1alpha1().NodeNetwork()
+	nad := management.CniFactory.K8s().V1().NetworkAttachmentDefinition()
+
+	handler := &Handler{
+		nodeNetworkCtr:   nns,
+		nodeNetworkCache: nns.Cache(),
+		nadCache:         nad.Cache(),
+		recorder:         management.NewRecorder(controllerName, "", ""),
+	}
+
+	nns.OnChange(ctx, controllerName, handler.OnChange)
+	nns.OnRemove(ctx, controllerName, handler.OnRemove)
+
+	// start network monitoring
+	go network.GetWatcher().Start(context.TODO())
+
+	// regular reset node network
+	go func() {
+		ticker := time.NewTicker(resetPeriod)
+		for range ticker.C {
+			klog.Infof("regular reset node network")
+			if err := handler.resetNodeNetwork(string(networkv1alpha1.NetworkTypeVLAN)); err != nil {
```

Comment:
\s\string(NetworkTypeVLAN)\NetworkTypeVLAN

Reply:
The function `resetNodeNetwork` will also be called in `pkg/network` package. It's better to pass parameter as string instead of `networkv1alpha1.NetworkType` to avoid importing `networkv1alpha1` package in `pkg/network`